lib/view/disclaimer_view.dart

```diff
@@ -1,206 +1,202 @@
-    child: const Text(
-      'You have already agreed 👍',
-      style: AppStyles.textH5,
-      textAlign: TextAlign.center,
-    ),
-  );
+  // When user agrees to disclaimer, persist values for that agreement, version of disclaimer and date/time to storage
+  Future<void> _setAgreed() async {
+    final prefs = await SharedPreferences.getInstance();
+    await prefs.setBool(Strings.settingDisclaimerAgreed, true);
+    await prefs.setString(Strings.settingDisclaimerVersion, Strings.disclaimerVersion);
+    await prefs.setString(Strings.settingDisclaimerAgreedDateTime, DateTime.now().toString());
+  }
 
-  // Floating container indicating to scroll down to agree
-  final _scrollDownToAgree = IgnorePointer(
-    child: Container(
-      decoration: const BoxDecoration(
-          color: AppColors.green50,
-          borderRadius: BorderRadius.all(Radius.circular(50))),
-      margin: const EdgeInsets.only(bottom: 16),
-      padding: const EdgeInsets.all(16),
-      child: const Text(
-        'Scroll down to agree',
-        style: AppStyles.textLegal,
-        textAlign: TextAlign.center,
-      ),
-    ),
-  );
+  // Check persisted values for disclaimer agreement if they exist
+  Future<DisclaimerDetails> _getAgreed() async {
+    final prefs = await SharedPreferences.getInstance();
+    final disclaimerValues = DisclaimerDetails();
 
-  // Floating container indicating to scroll down
-  final _scrollDown = IgnorePointer(
-    child: Container(
-      decoration: const BoxDecoration(
-          color: AppColors.green50,
-          borderRadius: BorderRadius.all(Radius.circular(50))),
-      margin: const EdgeInsets.only(bottom: 16),
-      padding: const EdgeInsets.all(16),
-      child: const Text(
-        'Scroll down',
-        style: AppStyles.textP,
-        textAlign: TextAlign.center,
-      ),
-    ),
-  );
+    disclaimerValues.agreed = prefs.getBool(Strings.settingDisclaimerAgreed) ?? false;
+    disclaimerValues.version = prefs.getString(Strings.settingDisclaimerVersion) ?? '0';
 
-  AnimationController _animationController;
-  Animation<double> _animation;
+    final dateStamp = prefs.getString(Strings.settingDisclaimerAgreedDateTime) ?? '';
+    if (dateStamp != '') {
+      disclaimerValues.dateStamp = DateFormat.yMMMd().add_jm().format(DateTime.parse(dateStamp));
+    }
 
-  @override
-  void initState() {
-    super.initState();
-    _animationController = AnimationController(
-      vsync: this,
-      duration: const Duration(milliseconds: 500),
-    );
-    _animation =
-        Tween<double>(begin: 1.0, end: 0.0).animate(_animationController);
+    return disclaimerValues;
   }
 
   @override
   Widget build(BuildContext context) {
-    // Bottom button to agree t&c
-    final _agreeButton = Container(
-      margin: const EdgeInsets.only(left: 16, right: 16, bottom: 16),
+    /// Agree button - button a user taps to agree to the disclaimer
+    final Widget _agreeButton = Container(
```

Comment:
I would convert this to a function like _agreeButton(BuildContext context) and move it outside the build method.

Reply:
Good idea 👍